### PR TITLE
feat: add indexing timestamp

### PR DESCRIPTION
**Pull request by lulf**
* Add a timestamp marking the time a document was indexed. This is generally useful to understand when a document was indexed both for debugging and for use by tests.
* Add unit test to verify the indexed timestamp.
* Add integration test to verify index is updated when updating a doc

### CODE REVIEW

File: integration-tests/tests/bombastic.rs [hunk 1]
```diff
@@ -174,6 +175,46 @@ async fn bombastic_search(context: &mut BombasticContext) {
     })
     .await;
 
+    // Update and check reindex
```
Comment (jcrossley3) [inline integration-tests/tests/bombastic.rs:178]: This test is getting a little big, no? Can we affirm the re-indexing in its own test that uploads an sbom twice, but doesn't do much else other than compare the timestamps?

File: integration-tests/tests/vexination.rs [hunk 2]
```diff
@@ -194,10 +194,10 @@ async fn test_upload_vex_empty_json(context: &mut VexinationContext) {
 
 #[test_context(VexinationContext)]
 #[tokio::test]
-#[ntest::timeout(60_000)]
+#[ntest::timeout(90_000)]
 async fn test_upload_vex_empty_file(vexination: &mut VexinationContext) {
     let id = "empty-file-upload";
-    wait_for_event(Duration::from_secs(30), &vexination.config, "vex-failed", id, async {
+    wait_for_event(Duration::from_secs(60), &vexination.config, "vex-failed", id, async {
```
Comment (jcrossley3) [inline integration-tests/tests/vexination.rs:200]: I'm curious why this is necessary?

**Review by jcrossley3 — APPROVED** [review]
Looks good, just a couple of questions

File: integration-tests/tests/vexination.rs (hunk 2 above)
Comment (lulf) [reply integration-tests/tests/vexination.rs:200]: It was failing in my initial PR build, and it looked like the timeout happened just before the indexer printed an error parsing the document. So either:

1) The timeout was to low
2) The indexer thread/task was blocked from running

Increasing the timeout should rule out option 1 at least.

File: integration-tests/tests/bombastic.rs (hunk 1 above)
Comment (lulf) [reply integration-tests/tests/bombastic.rs:178]: Agreed, I will move this and the 'invalid search' test cases into separate tests.

**Review by jcrossley3 — APPROVED** [review]
I dig the test refactorings. I'm doing some myself, so looking forward to the merge conflicts! ;)

**Pull request merged**